File: frontend/src/types/stowage.ts
export type CautionType = "decomposable" | "fragile" | "waste" | "edible" | "none";

export interface CargoInput {
  id: string;
  length: number;
  width: number;
  height: number;
  caution: CautionType;
}

export interface ShipInput {
  length: number;
  width: number;
  height: number;
}

export interface StowageRequest {
  ship: ShipInput;
  cargos: CargoInput[];
}

export interface CargoPlacement extends CargoInput {
  x: number;
  y: number;
  z: number;
}

export interface StowageResponse {
  placements: CargoPlacement[];
  utilization: number;
  result_id: number;
  report_url: string;
}

File: backend/app/schemas/stowage.py
# app/schemas/stowage.py
from pydantic import BaseModel
from typing import List, Literal

CautionType = Literal["decomposable", "fragile", "waste", "edible", "none"]

class CargoInput(BaseModel):
    id: str
    length: float
    width: float
    height: float
    caution: CautionType

class ShipInput(BaseModel):
    length: float
    width: float
    height: float

class StowageRequest(BaseModel):
    ship: ShipInput
    cargos: List[CargoInput]

class CargoPlacement(BaseModel):
    id: str
    x: float
    y: float
    z: float
    length: float
    width: float
    height: float
    caution: CautionType

class StowageResponse(BaseModel):
    placements: List[CargoPlacement]
    utilization: float
    result_id: int
    report_url: str
